src/haz3lcore/dynamics/DH.re

```diff
@@ -1,52 +1,55 @@
 open Sexplib.Std;
+open Hazel_menhir.AST;
 
 [@deriving (show({with_path: false}), sexp, yojson)]
 type if_consistency =
   | ConsistentIf
   | InconsistentIf;
 
 module rec DHExp: {
+  //All comments show the textual syntax for DHExp when built using the menhir parser
   [@deriving (show({with_path: false}), sexp, yojson)]
   type t =
-    | EmptyHole(MetaVar.t, HoleInstanceId.t)
-    | NonEmptyHole(ErrStatus.HoleReason.t, MetaVar.t, HoleInstanceId.t, t)
-    | FreeVar(MetaVar.t, HoleInstanceId.t, Var.t)
+    | EmptyHole(MetaVar.t, HoleInstanceId.t) //()
+    | NonEmptyHole(ErrStatus.HoleReason.t, MetaVar.t, HoleInstanceId.t, t) //(_HOLE x)
+    | FreeVar(MetaVar.t, HoleInstanceId.t, Var.t) //(_FREE x)
     | InvalidText(MetaVar.t, HoleInstanceId.t, string)
-    | InconsistentBranches(MetaVar.t, HoleInstanceId.t, case)
+    | InconsistentBranches(MetaVar.t, HoleInstanceId.t, case) //(_HOLE (case ...))
     | Closure([@opaque] ClosureEnvironment.t, t)
-    | Filter(DHFilter.t, t)
-    | BoundVar(Var.t)
-    | Sequence(t, t)
-    | Let(DHPat.t, t, t)
-    | FixF(Var.t, Typ.t, t)
-    | Fun(DHPat.t, Typ.t, t, option(Var.t))
+    | Filter(DHFilter.t, t) //pause a == 2 b
+    | BoundVar(Var.t) //x
+    | Sequence(t, t) //a; b
```

Comment:
let's use metavariables `e` for expressions, so `e1; e2` in the comment here (and similar elsewhere)